src/App.js

```diff
@@ -54,52 +56,79 @@ class App extends Component {
         let newChannel = {
           key: channel.id,
           value: channel.id,
-          text: channel.name
+          text: channel.name,
+          members: channel.members
         }
        return channels.push(newChannel);
       });
-      return this.setState({channels: channels});
+      return this.setState({ channels: channels });
     });
   }
   setGroupSize = (value) => {
-    return this.setState({groupSize: parseInt(value, 16)})
+    return this.setState({ groupSize: parseInt(value, 16) })
   }
-  buildGroups = (value) => {
+  selectedChannel = (value) => {
+    let channelKey = value.length? value : null;
+    return this.setState({
+      selectedChannel: channelKey,
+      useSpecificChannel: !this.state.useSpecificChannel
+     }, () => this.getChannelMembers());
+  }
+  getChannelMembers = () => {
+    let channelMembers = []
+    let selectedChannel = this.state.selectedChannel
+    let newUsers = []
+
+    if (this.state.channels && selectedChannel ) {
+      let channelObj = this.state.channels.find(function (channel) { return channel.value == selectedChannel });
+      channelMembers = channelObj.members
+
+      for (let i=0; i<channelMembers.length; i++){
+        var user = this.state.users.find(function (user) { return user.id === channelMembers[i]; });
+        // user !== undefined && newUsers.push(user); // this works, but is garbage for readability
+        if (user !== undefined){ // channels include admins and this accounts for return undefined
```

Comment:
Does this work? I think it should either be `if ( !(user) )` (catches null, undefined, or false), or `if (typeof user !== 'undefined')`

My vote would be the ` !(user)`